source/unified-test-format/unified-test-format.rst

```diff
@@ -0,0 +1,2534 @@
+use `$$matchesHexBytes`_ in `expectResult <operation_expectResult_>`_ to match
+the contents of the returned stream. These operations MAY use
+`saveResultAsEntity <operation_saveResultAsEntity_>`_ to save the stream for use
+with a subsequent operation (e.g. `uploadFromStream`_ ).
+
+
+.. _openUploadStream:
+.. _openUploadStreamWithId:
+
+openUploadStream and openUploadStreamWithId
+```````````````````````````````````````````
+
+These operations SHOULD NOT be used in test files. See
+`IO operations for GridFS streams`_ in `Future Work`_. 
+
+
+.. _uploadFromStream:
+.. _uploadFromStreamWithId:
+
+uploadFromStream and uploadFromStreamWithId
+```````````````````````````````````````````
+
+The ``uploadFromStream`` and ``uploadFromStreamWithId`` operations' ``stream``
+parameter is a stream as defined in the `GridFS <../gridfs/gridfs-spec.rst>`__
+spec and not easily expressed in YAML/JSON. This parameter is expressed as an
+entity name, which the test runner MUST resolve to a `stream <entity_stream_>`_
+entity *before* passing it as a parameter to the method. The YAML file SHOULD
+use an `alias node`_ for a stream entity's ``id`` field
+(e.g. ``stream: *stream0``).
+
+
+changeStream
+~~~~~~~~~~~~
+
+Change stream entities are special in that they are not defined in
+`createEntities`_ but are instead created by using
+`operation.saveResultAsEntity <operation_saveResultAsEntity_>`_ with a
+`client_createChangeStream`_, `database_createChangeStream`_, or
+`collection_createChangeStream`_ operation.
+
+Test files SHOULD NOT use a ``watch`` operation to create a change stream, as
+the implementation of that method may vary among drivers. For example, some
+implementations of ``watch`` immediately execute ``aggregate`` and construct the
+server-side cursor, while others may defer ``aggregate`` until the change stream
+object is iterated.
+
+The `Change Streams <../change-streams/change-streams.rst>`__ spec does not
+define a consistent API for the ChangeStream class, since the mechanisms for
+iteration and capturing a resume token may differ between synchronous and
+asynchronous drivers. To account for this, this section explicitly defines the
+supported operations for change stream entities.
+
+
+iterateUntilDocumentOrError
+```````````````````````````
+
+Iterates the change stream until either a single document is returned or an
+error is raised.
+
+If `expectResult <operation_expectResult_>`_ is specified, it SHOULD be a single
+document.
+
+`Iterating the Change Stream <../change-streams/tests#iterating-the-change-stream>`__
+in the change stream spec cautions drivers that implement a blocking mode of
+iteration (e.g. asynchronous drivers) not to iterate the change stream
+unnecessarily, as doing so could cause the test runner to block indefinitely.
+This should not be a concern for ``iterateUntilDocumentOrError`` as iteration
+only continues until either a document or error is encountered.
+
+
+Special Test Operations
```

Comment:
Will this spec include the SDAM integration test operations? https://github.com/mongodb/specifications/tree/master/source/server-discovery-and-monitoring/tests#special-test-operations

Reply:
FLE and SDAM integration tests were never in scope ([WRITING-4394](https://jira.mongodb.org/browse/WRITING-4394)), so I did not intend to add them in the initial schema version. It would be good to confirm that their support would just be additive changes (e.g. 1.1) without necessitating BC breaks. Things like new operation types (entity or special) or extra options when creating an entity (e.g. FLE autoEncryptOptions) can be considered additive.

If their support _would_ entail BC breaks, then I think it'd make sense to make the minimum number of changes to this initial format and avoid that. By "minimum number of changes", I mean restructure some of the syntax that might conflict (e.g. the way we do event assertions), rather than adding all of the new operations.

Reply:
Thanks I think all the SDAM test changes would be additive.